sdk/keyvault/azure-security-keyvault-keys/src/list_keys_single_page_result.cpp

```diff
@@ -125,3 +127,33 @@ DeletedKeySinglePage _detail::KeyPropertiesSinglePageSerializer::DeletedKeySingl
 
   return deletedKeySinglePage;
 }
+
+void DeletedKeySinglePage::OnNextPage(const Azure::Core::Context& context)
+{
+  // Before calling `OnNextPage` pagedResponse validates there is a next page, so we are sure
+  // NextPageToken is valid.
+  GetDeletedKeysSinglePageOptions options;
+  options.NextPageToken = NextPageToken;
+  *this = keyClient->GetDeletedKeysSinglePage(options, context);
+  CurrentPageToken = options.NextPageToken.Value();
+}
+
+void KeyPropertiesSinglePage::OnNextPage(const Azure::Core::Context& context)
+{
+  // Before calling `OnNextPage` pagedResponse validates there is a next page, so we are sure
+  // NextPageToken is valid.
+  if (m_type == KeyPropertiesSinglePage::KeyPropertiesType::Keys)
+  {
+    GetPropertiesOfKeysSinglePageOptions options;
+    options.NextPageToken = NextPageToken;
+    *this = m_keyClient->GetPropertiesOfKeysSinglePage(options, context);
+    CurrentPageToken = options.NextPageToken.Value();
+  }
+  else if (m_type == KeyPropertiesSinglePage::KeyPropertiesType::Versions)
```

Comment:
What happens if the `m_type` is not one of these? Abort in an `else` case just in case, if that would be an unexpected state.

Reply:
I have removed this extra type. I can directly use the keyName string a the way to choose between listing keys or key's versions